x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -272,11 +306,19 @@ public void lookupUser(final String username, final ActionListener<User> listene
     @Override
     public void expire(final String username) {
         this.ensureInitialized();
+        LOGGER.trace("Expiring JWT cache entries for realm [" + super.name() + "] principal=[" + username + "]");
+        this.jwtCacheHelper.removeValuesIf(expiringUser -> expiringUser.user.principal().equals(username));
```

Comment:
Ok. I think it makes sense to add a null check in `CacheIteratorHelper`. Since `removeValuesIf` is a copy of `removeKeysIf`, that implies both need a null check.

https://github.com/elastic/elasticsearch/blob/45f2851562e68c393954df5be942e1764e8dbe09/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/CacheIteratorHelper.java#L48-L59

https://github.com/elastic/elasticsearch/blob/45f2851562e68c393954df5be942e1764e8dbe09/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/CacheIteratorHelper.java#L61-L72